### PR TITLE
added a dye builder (example in CraftyItems)

### PR DESCRIPTION
**Pull request by ItsArt3x54**
I added a DyeBuilder, its like the ItemBuilder but with 3 news methods.

### CODE REVIEW

File: src/main/java/be/raft/crafty/item/DyeBuilder.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /**
+     * Set the color of the dye
+     * @param color color
+     */
+    public DyeBuilder setColor(Color color) {
+        return this.setColor(color.getData());
+    }
+
+    /**
+     * Set the color of the dye with a byte
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/DyeBuilder.java]: Could you change this line to "Set the color of the dye **using** a byte"

File: src/main/java/be/raft/crafty/item/PotionBuilder.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+public class PotionBuilder extends ItemBuilder {
+    protected PotionBuilder(ItemStack stack, PotionMeta meta) {
+        super(stack, meta);
+    }
+
+    /**
+     * Adds a custom potion effect to this potion
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/PotionBuilder.java]: Missing Ponctuation please use '.' at the end of your phrase.

File: src/main/java/be/raft/crafty/item/PotionBuilder.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+     * @param potionEffect potion effect
+     */
+    public PotionBuilder addCustomPotionEffect(PotionEffect potionEffect) {
+        return this.addCustomPotionEffect(potionEffect);
+    }
+
+    /**
+     * Adds a custom potion effect to this potion
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/PotionBuilder.java]: Missing Ponctuation please use '.' at the end of your phrase.

File: src/main/java/be/raft/crafty/item/PotionBuilder.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+     * @param potionEffect potion effect
+     * @param overwrite true if any existing effect of the same type should be overwritten
+     */
+    public PotionBuilder addCustomPotionEffect(PotionEffect potionEffect, boolean overwrite) {
+        return this.addCustomPotionEffect(potionEffect, overwrite);
+    }
+
+    /**
+     * Removes a potion effect from the potion
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/PotionBuilder.java]: Missing Ponctuation please use '.' at the end of your phrase.

File: src/main/java/be/raft/crafty/item/PotionBuilder.java [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+     * @param type potion effect type
+     */
+    public PotionBuilder removeCustomPotionEffect(PotionEffectType type) {
+        return this.removeCustomPotionEffect(type);
+    }
+
+    /**
+     * Moves a potion effect to the top of the potion effect list
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/PotionBuilder.java]: Missing Ponctuation please use '.' at the end of your phrase.

**Review by BelgianDev — CHANGES_REQUESTED** [review]
Big Thanks for the contribution! I reviewed your code and added some things to change before I can merge it.

File: src/main/java/be/raft/crafty/item/PotionBuilder.java [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+
+    /**
+     * Create a Builder from a defined {@link Material}.
+     * @param material material
+     */
+    public static PotionBuilder create(Material material) {
+        return PotionBuilder.create(new ItemStack(material));
+    }
+
+}
```
Comment (BelgianDev) [inline src/main/java/be/raft/crafty/item/PotionBuilder.java:69]: Are you sure this is even working? All these function only call themself. Did you test that code

**Comment by ItsArt3x54** [comment]
I hadn't test the plugin yet, I see errors i'm fixing it tomorow

**Comment by BelgianDev** [comment]
Please test your code before creating a pull request.

**Comment by ItsArt3x54** [comment]
I found a bug with the potions tho, the effect aren't listed in the lore. But i made the itemstack without PotionBuilder and it has the same bug so i think that it is from bukkit.

**Comment by BelgianDev** [comment]
It's not a bug but a feature, so no issues here.

**Comment by ItsArt3x54** [comment]
> It's not a bug but a feature, so no issues here.

So why does the setMainEffect exist?

**Review by BelgianDev — APPROVED** [review]

**Comment by BelgianDev** [comment]
Alright, Looks good for me. Just let the checks do their job before I merge it.

**Pull request merged**